src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2386,6 +2438,218 @@ object Desugar {
 
+    /**
+      * Generates proof obligations for the initialization of the package under verification.
+      * @param mainPkg package under verification
+      * @param specCollector info about the init specifications of imported packages. All imported packages should
+      *                      have been registered in `specCollector` before calling this method
+      * @param config
+      */
+    def registerMainPackage(mainPkg: PPackage, specCollector: PackageInitSpecCollector)(config: Config): Unit = {
+      // As for imported methods, imported packages are not checked to satisfy their specification. In particular,
+      // Gobra does not check that the package postconditions are established by the initialization code. Instead,
+      // this is assumed. We only generate the proof obligations for the initialization code in the main package.
+      registerPackage(mainPkg, specCollector, generateInitProof = true)(config)
+
+      // check that the postconditions of every package are enough to satisfy all of their imports' preconditions
+      val src = meta(mainPkg, info)
+      specCollector.registeredPackages().foreach{ pkg =>
+        val checkPkgImports = checkPkgImportObligations(pkg, specCollector)(src)
+        AdditionalMembers.addMember(checkPkgImports)
+      }
+
+      // proof obligations for function main
+      val mainFuncObligations = generateMainFuncObligations(mainPkg, specCollector)
+      mainFuncObligations.foreach(AdditionalMembers.addMember)
+    }
+
+    /**
+      * Generates the members that correspond to the global variables declared in `pkg`
+      * and registers info in `specCollector` about all import preconditions and
+      * all package postconditions in all files of `pkg`.
+      * @param pkg
+      * @param specCollector
+      * @param generateInitProof true if the proof obligations for the package's initialization code
+      *                          should be generated
+      * @param config
+      */
+    def registerPackage(pkg: PPackage, specCollector: PackageInitSpecCollector, generateInitProof: Boolean = false)
+                       (config: Config): Unit = {
+      // Desugar all declarations of globals in `pkg` to generate the members that correspond to the global variables.
+      // Each entry in `pGlobalDecls` contains the declarations of a file in `pkg` sorted by the order in which they
+      // appear in the program.
+      val pGlobalDecls: Vector[Vector[PVarDecl]] = pkg.programs.map{ p =>
+        val unsortedDecls = p.declarations.collect{ case d: PVarDecl => d }
+        // TODO: this is currently fine, given that we currently require global variable declarations to come after
+        //       the declarations of all of the dependencies of the declared variables. This should be changed when
+        //       we lift this restriction.
+        unsortedDecls.sortBy{ decl =>
```

Comment:
I could move it to `PProgram`, but I would need to provide a `PositionManager` as a parameter to the getter method. Also, one of the occurrences is meant to be temporary, i.e., it occurs just because we enforce a temporary check via method the `globalDeclSatisfiesDepOrder` in `ProgramTyping` (the idea is to drop it in the PR where Gobra is extended to support variables declared out of order). Do you think this is still worth doing?